components/tikv_alloc/src/trace.rs

```diff
@@ -237,6 +237,77 @@ macro_rules! mem_trace {
     }
 }
 
+pub struct MemTraced<T: Default> {
+    item: T,
+    size: usize,
+    node: Option<Arc<dyn MemoryTrace + Send + Sync>>,
```

Comment:
How about extracting these fields into a `Guard`?

My thought: `Guard` is built via `MemoryTrace::guard(size)`. We can do `MemoryTrace::trace(Add)` inside `MemoryTrace::guard` and `MemoryTrace::trace(Sub)` inside `Guard::drop`.